### PR TITLE
NIAD-1107: Consistently handle unresolvable references in Mappings

### PR DESCRIPTION
**Pull request by bbell-kainos**

### CODE REVIEW

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/InputBundle.java [hunk 1]
```diff
@@ -64,7 +64,11 @@ public InputBundle(Bundle bundle) {
         var resourceType = ResourceType.fromCode(reference.getResourceType());
 
         if (MAPPABLE_RESOURCES.contains(resourceType)) {
-            return extractResourceByReference(this.bundle, reference);
+            Optional<Resource> resource = extractResourceByReference(this.bundle, reference);
+            if (resource.isPresent()) {
+                return resource;
+            }
+            throw new EhrMapperException("Resource not found: " + reference);
```
Comment (rorysproule1) [inline service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/InputBundle.java:71]: Are we sure this should be changed from Reference?

File: service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/RequestStatementMapperTest.java [hunk 2]
```diff
@@ -259,9 +259,9 @@
             arguments(INPUT_JSON_WITH_INCORRECT_RESOURCE_TYPE_REQUESTER, "Requester Reference not of expected Resource Type"),
             arguments(INPUT_JSON_WITH_INCORRECT_RESOURCE_TYPE_RECIPIENT, "Recipient Reference not of expected Resource Type"),
             arguments(INPUT_JSON_WITH_INCORRECT_RESOURCE_TYPE_AUTHOR, "Author Reference not of expected Resource Type"),
-            arguments(INPUT_JSON_WITH_NO_RESOLVED_REFERENCE_REQUESTER, "Could not resolve Device Reference"),
-            arguments(INPUT_JSON_WITH_NO_RESOLVED_REFERENCE_RECIPIENT, "Could not resolve Organization Reference"),
-            arguments(INPUT_JSON_WITH_NO_RESOLVED_REFERENCE_NOTE_AUTHOR, "Could not resolve RelatedPerson Reference"),
+            arguments(INPUT_JSON_WITH_NO_RESOLVED_REFERENCE_REQUESTER, "Resource not found: Device/un-resolved"),
```
Comment (rorysproule1) [inline service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/RequestStatementMapperTest.java:262]: Same again really, just confirm this use of resource over reference

**Review by rorysproule1 — APPROVED** [review]

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/InputBundle.java (hunk 1 above)
Comment (bbell-kainos) [reply service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/InputBundle.java:71]: Following conversation with @dmcavenue1, 'Resource not found' is acceptable

File: service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/RequestStatementMapperTest.java (hunk 2 above)
Comment (bbell-kainos) [reply service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/RequestStatementMapperTest.java:262]: See earlier comment

File: docker/wiremock/stubs/__files/correctPatientStructuredRecordResponse.json [hunk 3]
```diff
@@ -16453,6 +16453,12 @@
                     }
                 }
             }
+        },
```
Comment (ElliotB-96) [inline docker/wiremock/stubs/__files/correctPatientStructuredRecordResponse.json:16456]: Well figured out.

**Review by ElliotB-96 — APPROVED** [review]
Looks good

**Pull request merged**